src/Symfony/Component/Messenger/Middleware/HandleMessageMiddleware.php

```diff
@@ -66,6 +78,21 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
             $this->logger->info('No handler for message "{class}"', $context);
         }
 
+        if (\count($exceptions)) {
+            throw new ChainedHandlerFailedException($envelope, ...$exceptions);
+        }
+
         return $stack->next()->handle($envelope, $stack);
     }
+
+    private function hasMessageSeen(Envelope $envelope, callable $handler, ?string $alias): bool
+    {
+        $some = array_filter($envelope
```

Comment:
You could use `array_find` instead actually, it would be faster because would stop at the first finding.

Reply:
What `array_find`? There is non in the standard lib

Reply:
😂 my god, good point. `array_search`.